presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -80,24 +85,36 @@ public PlanNodeStatsEstimate calculateStats(PlanNode node, StatsProvider sourceS
     }
 
     @Override
-    public void registerPlan(PlanNode root, Session session)
+    public boolean registerPlan(PlanNode root, Session session, long startTimeInNano, long timeoutInMilliseconds)
     {
         // Only precompute history based stats when plan has a join/aggregation.
-        if (!PlanNodeSearcher.searchFrom(root).where(node -> PRECOMPUTE_PLAN_NODES.stream().anyMatch(clazz -> clazz.isInstance(node))).matches()) {
-            return;
+        if (!prefetchForAllPlanNodes && !PlanNodeSearcher.searchFrom(root).where(node -> PRECOMPUTE_PLAN_NODES.stream().anyMatch(clazz -> clazz.isInstance(node))).matches()) {
+            return false;
```

Comment:
If we only care about query plans with join/aggregation, return false here, and skip adding stats equivalent node for this query plan.